### ISSUE TITLE
MySQLConnector 1.2 is not working as an AdoNet provider

### ISSUE THREAD

**Issue by emilekberg** [dotnet/orleans#6852]
When going to a major version, MySQLConnector seems to have made changes to the namespace, thus making the current MySQLConnector adonet provider not working.  https://github.com/mysql-net/MySqlConnector/releases/tag/1.0.0

It seems the Invariant tuple needs to be updated.
```
{ AdoNetInvariants.InvariantNameMySqlConnector, new Tuple<string, string>("MySqlConnector", "MySql.Data.MySqlClient.MySqlClientFactory") },
```

the error in logs can be seen as

```
Orleans.Runtime.SiloLifecycleSubject - MembershipTableManager failed to start due to errors at stage RuntimeGrainServices (8000): System.InvalidOperationException: Unable to load type 'MySql.Data.MySqlClient.MySqlClientFactory' for 'MySql.Data.MySqlConnector' invariant name.
   at Orleans.Clustering.AdoNet.Storage.DbConnectionFactory.GetFactory(String invariantName)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Orleans.Clustering.AdoNet.Storage.DbConnectionFactory.CreateConnection(String invariantName, String connectionString)
   at Orleans.Clustering.AdoNet.Storage.RelationalStorage.ExecuteAsync[TResult](String query, Action`1 parameterProvider, Func`5 executor, Func`4 selector, CancellationToken cancellationToken, CommandBehavior commandBehavior)
   at Orleans.Clustering.AdoNet.Storage.RelationalStorage.ReadAsync[TResult](String query, Action`1 parameterProvider, Func`4 selector, CancellationToken cancellationToken, CommandBehavior commandBehavior)
   at Orleans.Clustering.AdoNet.Storage.RelationalOrleansQueries.CreateInstance(String invariantName, String connectionString, IGrainReferenceConverter grainReferenceConverter)
   at Orleans.Runtime.MembershipService.AdoNetClusteringTable.InitializeMembershipTable(Boolean tryInitTableVersion)
   at Orleans.Runtime.MembershipService.MembershipTableManager.Start()
   at Orleans.Runtime.MembershipService.MembershipTableManager.<>c__DisplayClass53_0.<<Orleans-ILifecycleParticipant<Orleans-Runtime-ISiloLifecycle>-Participate>g__OnRuntimeGrainServicesStart|0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Orleans.Runtime.SiloLifecycleSubject.MonitoredObserver.OnStart(CancellationToken ct)
20201208 15:10:27.383 | E | Orleans.Runtime.SiloLifecycleSubject - Lifecycle start canceled due to errors at stage 8000: System.InvalidOperationException: Unable to load type 'MySql.Data.MySqlClient.MySqlClientFactory' for 'MySql.Data.MySqlConnector' invariant name.
   at Orleans.Clustering.AdoNet.Storage.DbConnectionFactory.GetFactory(String invariantName)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Orleans.Clustering.AdoNet.Storage.DbConnectionFactory.CreateConnection(String invariantName, String connectionString)
   at Orleans.Clustering.AdoNet.Storage.RelationalStorage.ExecuteAsync[TResult](String query, Action`1 parameterProvider, Func`5 executor, Func`4 selector, CancellationToken cancellationToken, CommandBehavior commandBehavior)
   at Orleans.Clustering.AdoNet.Storage.RelationalStorage.ReadAsync[TResult](String query, Action`1 parameterProvider, Func`4 selector, CancellationToken cancellationToken, CommandBehavior commandBehavior)
   at Orleans.Clustering.AdoNet.Storage.RelationalOrleansQueries.CreateInstance(String invariantName, String connectionString, IGrainReferenceConverter grainReferenceConverter)
   at Orleans.Runtime.MembershipService.AdoNetClusteringTable.InitializeMembershipTable(Boolean tryInitTableVersion)
   at Orleans.Runtime.MembershipService.MembershipTableManager.Start()
   at Orleans.Runtime.MembershipService.MembershipTableManager.<>c__DisplayClass53_0.<<Orleans-ILifecycleParticipant<Orleans-Runtime-ISiloLifecycle>-Participate>g__OnRuntimeGrainServicesStart|0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Orleans.Runtime.SiloLifecycleSubject.MonitoredObserver.OnStart(CancellationToken ct)
   at Orleans.LifecycleSubject.OnStart(CancellationToken ct)
20201208 15:10:27.388 | E | Orleans.Runtime.Silo - Exception during Silo.Start
System.InvalidOperationException: Unable to load type 'MySql.Data.MySqlClient.MySqlClientFactory' for 'MySql.Data.MySqlConnector' invariant name.
   at Orleans.Clustering.AdoNet.Storage.DbConnectionFactory.GetFactory(String invariantName)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Orleans.Clustering.AdoNet.Storage.DbConnectionFactory.CreateConnection(String invariantName, String connectionString)
   at Orleans.Clustering.AdoNet.Storage.RelationalStorage.ExecuteAsync[TResult](String query, Action`1 parameterProvider, Func`5 executor, Func`4 selector, CancellationToken cancellationToken, CommandBehavior commandBehavior)
   at Orleans.Clustering.AdoNet.Storage.RelationalStorage.ReadAsync[TResult](String query, Action`1 parameterProvider, Func`4 selector, CancellationToken cancellationToken, CommandBehavior commandBehavior)
   at Orleans.Clustering.AdoNet.Storage.RelationalOrleansQueries.CreateInstance(String invariantName, String connectionString, IGrainReferenceConverter grainReferenceConverter)
   at Orleans.Runtime.MembershipService.AdoNetClusteringTable.InitializeMembershipTable(Boolean tryInitTableVersion)
   at Orleans.Runtime.MembershipService.MembershipTableManager.Start()
   at Orleans.Runtime.MembershipService.MembershipTableManager.<>c__DisplayClass53_0.<<Orleans-ILifecycleParticipant<Orleans-Runtime-ISiloLifecycle>-Participate>g__OnRuntimeGrainServicesStart|0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Orleans.Runtime.SiloLifecycleSubject.MonitoredObserver.OnStart(CancellationToken ct)
   at Orleans.LifecycleSubject.OnStart(CancellationToken ct)
   at Orleans.Runtime.Scheduler.AsyncClosureWorkItem.Execute()
   at Orleans.Runtime.Silo.StartAsync(CancellationToken cancellationToken)
```

Downgrading to the version 0.69.10 sorts this (as the namespace changes).

**Comment by srollinet** [comment]
I've seen there is a PR for this problem: https://github.com/dotnet/orleans/pull/6831

**Comment by emilekberg** [comment]
Tried quickly with 3.4.0 and this is now solved with #6831. Thanks for fixing this!
